src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -543,6 +544,62 @@ static bool simplify_reduction(shared_ptr<Node> n)
     return true;
 }
 
+static bool replace_transpose_with_reshape(shared_ptr<Node> n)
+{
+    bool rc = false;
+    auto transpose = as_type_ptr<op::Transpose>(n);
+
+    if (transpose)
+    {
+        auto perm = as_type_ptr<op::Constant>(n->get_argument(1));
+        auto perm_value = perm->get_vector<int64_t>();
+        auto data = n->get_argument(0);
+        auto data_shape = n->input_value(0).get_partial_shape().to_shape();
+
+        // Check if input shape contains a Dim of value 1
```

Comment:
Why not just run through the perm axes and if those dimensions are all static of length 1 then you can do the reshape.